### PR TITLE
Update swagger-codegen-maven-plugin example to use current dependencies

### LINKED ISSUE #3261: Example swagger-codegen-maven-plugin/examples/java-client.xml is out of date
Example dependencies from [swagger-codegen-maven-plugin/examples/java-client.xml](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen-maven-plugin/examples/java-client.xml) do not work correctly.

I had to use the generation webservice to extract a pom with the proper dependencies for jersey2

They should be changed to something like this if they are indeed the best practices:

``` xml
<!-- HTTP client: jersey-client -->
    <dependency>
      <groupId>org.glassfish.jersey.core</groupId>
      <artifactId>jersey-client</artifactId>
      <version>${jersey-version}</version>
    </dependency>
    <dependency>
      <groupId>org.glassfish.jersey.media</groupId>
      <artifactId>jersey-media-multipart</artifactId>
      <version>${jersey-version}</version>
    </dependency>
    <dependency>
      <groupId>org.glassfish.jersey.media</groupId>
      <artifactId>jersey-media-json-jackson</artifactId>
      <version>2.22.1</version>
    </dependency>

    <!-- JSON processing: jackson -->
    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-core</artifactId>
      <version>${jackson-version}</version>
    </dependency>
    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-annotations</artifactId>
      <version>${jackson-version}</version>
    </dependency>
    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-databind</artifactId>
      <version>${jackson-version}</version>
    </dependency>
    <dependency>
      <groupId>com.fasterxml.jackson.datatype</groupId>
      <artifactId>jackson-datatype-joda</artifactId>
      <version>2.1.5</version>
    </dependency>
    <dependency>
      <groupId>joda-time</groupId>
      <artifactId>joda-time</artifactId>
      <version>${jodatime-version}</version>
    </dependency>

    <!-- Base64 encoding that works in both JVM and Android -->
    <dependency>
      <groupId>com.brsanthu</groupId>
      <artifactId>migbase64</artifactId>
      <version>2.2</version>
    </dependency>

    <!-- test dependencies -->
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>${junit-version}</version>
      <scope>test</scope>
    </dependency>
  </dependencies>
  <properties>
    <swagger-core-version>1.5.8</swagger-core-version>
    <jersey-version>2.22.2</jersey-version>
    <jackson-version>2.7.0</jackson-version>
    <jodatime-version>2.9.3</jodatime-version>
    <maven-plugin-version>1.0.0</maven-plugin-version>
    <junit-version>4.12</junit-version>
  </properties>
```

### PR DESCRIPTION
**Pull request by Trii**
Fixes #3261 
New PR for history vanity

### CODE REVIEW

File: modules/swagger-codegen-maven-plugin/examples/java-client.xml [hunk 1]
```diff
@@ -101,15 +96,16 @@
...
-            <scope>test</scope>
+            <!--<scope>test</scope>-->
         </dependency>
     </dependencies>
     <properties>
         <swagger-annotations-version>1.5.0</swagger-annotations-version>
-        <jersey-version>2.12</jersey-version>
-        <jackson-version>2.4.2</jackson-version>
-        <jodatime-version>2.3</jodatime-version>
+        <swagger-core-version>1.5.8</swagger-core-version>
```
Comment (wing328) [inline modules/swagger-codegen-maven-plugin/examples/java-client.xml]: for swagger core, i believe the latest stable version is 1.5.9: https://mvnrepository.com/artifact/io.swagger/swagger-core

Comment (Trii) [inline modules/swagger-codegen-maven-plugin/examples/java-client.xml]: I used the output from https://generator.swagger.io/#!/clients/generateClient in order to get the version numbers. That generator API returns 1.5.8 so I copied it in order to stay on par with what was already being done.

I can update it if you think there are no breaking changes between versions.

**Comment by wing328** [comment]
generator.swagger.io uses the latest stable version (2.1.6) but not the latest master. After the 2.1.16 release, we've reviewed and updated the dependencies for all langauges so please use 1.5.9 instead for swagger core version.

**Comment by wing328** [comment]
cc @ePaul

**Comment by Trii** [comment]
Where can I find that list in code so I can match it up in the future?

**Comment by wing328** [comment]
For Java dependencies, I usually go to https://mvnrespository.com to look for the latest stable version.

e..g https://mvnrepository.com/artifact/io.swagger/swagger-core

**Comment by Trii** [comment]
I meant which versions you know that the generated templates work with. I
don't want to blindly bump to latest if that doesn't work with the
generated Java files.

**Comment by wing328** [comment]
The template should work with the latest stable version. If there's any issues, we definitely need to look into those.

If you do not feel comfortable upgrading swagger core to 1.5.9, it's perfectly to fine to keep 1.5.8 and we'll do the upgrade later.

**Comment by Trii** [comment]
@wing328 it is building now. It would be nice to have something in the build process to automate these dependency updates so they stay in sync automatically

**Comment by wing328** [comment]
@Trii agree that automating the dependency updates would be nice.

For the CI failure, it's due to https://github.com/npm/npm/issues/13284, which has been ongoing for hours...

I'll restart the CI jobs once the issue has been addressed.

**Comment by wing328** [comment]
There was an issue reported by the CI as follows:
```
[0K$ cd modules/swagger-codegen-maven-plugin/examples && mvn -f java-client.xml swagger-codegen:generate
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building sample-project 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- swagger-codegen-maven-plugin:2.2.0-SNAPSHOT:generate (default-cli) @ sample-project ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.902s
[INFO] Finished at: Thu Jul 07 04:06:01 UTC 2016
[INFO] Final Memory: 10M/142M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.swagger:swagger-codegen-maven-plugin:2.2.0-SNAPSHOT:generate (default-cli) on project sample-project: The parameters 'inputSpec', 'language' for goal io.swagger:swagger-codegen-maven-plugin:2.2.0-SNAPSHOT:generate are missing or invalid -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginParameterException
```

@Trii I wonder if you can take a look when you've time.

File: modules/swagger-codegen-maven-plugin/examples/java-client.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@
             <plugin>
                 <groupId>io.swagger</groupId>
                 <artifactId>swagger-codegen-maven-plugin</artifactId>
-                <version>2.1.5-SNAPSHOT</version>
+                <version>2.2.0-SNAPSHOT</version>
```
Comment (wing328) [inline modules/swagger-codegen-maven-plugin/examples/java-client.xml]: 2.2.0 has been released: https://mvnrepository.com/artifact/io.swagger/swagger-codegen-maven-plugin

Please kindly remove the `-SNAPSHOT`

**Comment by wing328** [comment]
@Trii  I tested with the latest master and got the same error when running `mvn -f java-client.xml swagger-codegen:generate`

Does the command run without issue in your local environment?

**Comment by Trii** [comment]
@wing328 `mvn -f java-client.xml swagger-codegen:generate` has never worked. The only way I have been able to get it to work was by moving the `inputSpec` parameter up one level in the `java-client.xml` as it doesn't seem to process those parameters when running the `swagger-codegen:generate` goal directly.

**Comment by wing328** [comment]
@Trii thanks for the info. Can you please update the java-client.xml as well with the fix? Otherwise, there's no way the CI tests can pass when running the command.

**Comment by Trii** [comment]
@wing328 honestly I quit fighting with it. The code generation library works well but the maven plugin is not quite right. The codegen tool also generates test stubs but the maven plugin does not. I'm not a java developer so it is out of the scope of my skill set at the moment. I am writing more java lately, but my deliverables preclude any extra curricular work I can put into this plugin at the moment. 

As I get more familiar with how maven plugins work, I can probably come back and revisit this item.

**Comment by wing328** [comment]
@Trii thanks again for your time and effort in fixing it. I'll try to get a core team member to address that.

**Comment by Trii** [comment]
@wing328 I may be able to revisit this in a week or two. I'll keep you updated.

**Comment by dyeshurun** [comment]
Any update about this? I keep getting the error above (using 2.2.3)

**Comment by frantuma** [comment]
replaced by #8448

**Pull request closed without merging**